### PR TITLE
fix: apply `_field` restrictions that do not match any fields

### PR DESCRIPTION
**Pull request by alamb**
Fixes https://github.com/influxdata/conductor/issues/891

# Rationale:

A predicate like `_field = 'foo'` means restrict to only data that has the field named `foo`. This was being ignored by IOx when ruling out chunks, resulting in incorrect results.

# Changes:
Properly evaluate the `_field` predicates

# TODO
- [x] verify that the reported test case passes with this fix
- [x] Update internal CI test that was skipping this test (PR is up)

### CODE REVIEW

File: db/src/access.rs [hunk 1]
```diff
@@ -232,6 +232,25 @@ impl ChunkAccess {
             .catalog_snapshot_duration
             .inc(start.elapsed());
 
+        // if there is a field restriction on the predicate, only
+        // chunks with that field should be returned. If the chunk has
+        // none of the fields specified, then it doesn't match
+        let chunks = match &predicate.field_columns {
```
Comment (alamb) [inline db/src/access.rs:238]: this is the fix

File: query_tests/src/influxrpc/tag_values.rs [hunk 2]
```diff
@@ -332,6 +332,46 @@ async fn list_tag_values_field_col() {
...
+#[tokio::test]
+async fn list_tag_values_field_col_does_not_exist() {
+    let tag_name = "state";
+    let predicate = PredicateBuilder::default()
+        .timestamp_range(0, 1000) // get all rows
+        // since this field doesn't exist this predicate should match no values
+        .add_expr(col("_field").eq(lit("not_a_column")))
+        .build();
+    let predicate = InfluxRpcPredicate::new(None, predicate);
+    let expected_tag_keys = vec![];
```
Comment (alamb) [inline query_tests/src/influxrpc/tag_values.rs:344]: this test fails (incorrectly returns `MA`, `CA`, and `NY`)

Comment (onelson) [reply query_tests/src/influxrpc/tag_values.rs:344]: This test is a nice explanation of the problem. Huh.

**Review by onelson — APPROVED** [review]
I don't really know enough of the context to vet the change, but if the tests are green this LGTM :shipit:

File: db/src/access.rs [hunk 3]
```diff
@@ -233,6 +233,25 @@ impl ChunkAccess {
...
+            None => chunks,
+            Some(field_columns) => {
+                chunks
```
Comment (tustvold) [inline db/src/access.rs:242]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain might be a nicer way to write this

**Review by tustvold — APPROVED** [review]
I'm not entirely sure how a bug in chunk pruning would cause more data to be returned that expected, if anything this feels like it is masking an underlying bug, but this code looks correct to me

**Comment by alamb** [comment]
So I thought more about this. Part of the confusion is that the InfluxRPC uses predicates for a concept that is very different in SQL: selection

`_field = 'foo'` means "select only the `foo` column. However, if there is no `foo` column there should be no columns (or rows) selected. So I guess we could also add a `false` Expr to the `Predicate` if there were no columns. But I am having a hard time coming up with a reproducer for that case.

**Pull request merged**